src/grpc_server.cc

```diff
@@ -3681,9 +3681,11 @@ ModelInferHandler::Process(InferHandler::State* state, bool rpc_ok)
     }
     // Get request ID for logging in case of error.
     const char* request_id = nullptr;
-    LOG_TRITONSERVER_ERROR(
-        TRITONSERVER_InferenceRequestId(irequest, &request_id),
-        "unable to retrieve request ID string");
+    if (irequest != nullptr) {
```

Comment:
Should this return if the `irequest` is null? The `request_id` becomes `"<id_unknown>"` but then we will attempt to use the null `irequest` on L3704  for TRITONSERVER_ServerInferAsync.

Reply:
I don't think TRITONSERVER_ServerInferAsync on L3704 will be triggered since the error is set on L3628 and we have this check on L3692